corehq/apps/reports/tests/test_case_list_explorer_report.py

```diff
@@ -0,0 +1,82 @@
+import json
+from django.http.request import QueryDict
+from django.test import TestCase
+from django.test.client import RequestFactory
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.reports.standard.cases.case_list_explorer import CaseListExplorer
+from corehq.apps.users.models import (
+    DomainMembership,
+    WebUser,
+)
+from corehq.util.test_utils import flag_enabled
+
+
+class TestCaseListExplorer(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_name = 'case-list-test'
+        cls.domain = Domain.get_or_create_with_name(cls.domain_name, is_active=True)
+        cls.user = WebUser(username='test@cchq.com', domains=[cls.domain_name], is_admin=True)
+        cls.user.domain_memberships.append(DomainMembership(domain=cls.domain_name, role_id='admin'))
+        cls.request_factory = RequestFactory()
+
+    def setUp(self):
+        super().setUp()
+        self.request = self.request_factory.get('/some/url')
+        self.request.couch_user = self.user
+        self.request.domain = self.domain
+        self.request.can_access_all_locations = True
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+
+    @flag_enabled('CASE_LIST_EXPLORER')
+    def test_with_explorer_columns_legacy(self):
+        legacy_columns = ['@case_type', 'case_name', 'last_modified']
+        q_dict_get = QueryDict('', mutable=True)
```

Comment:
nit: short names like this, do add an extra overhead later for anyone looking at the code. Just `query_dict` or `get_query_dict` would make it clear.